### DISCUSSION TITLE
Report - (4th Sem.) Week 2 📢

### DISCUSSION THREAD

**Discussion by denniskp** [SE-TINF22B6/DHBWhub#528] [Weekly Report]
Greetings and welcome to our tenth progress report! We're thrilled to update you on the latest news and developments since the beginning of our seventh sprint! 🥳

#### New mockups
This week, our team dedicated efforts to crafting a fresh mockup for the friends and FAQ page, aligning it with our evolving design goals.

> ![friends](https://github.com/SE-TINF22B6/DHBWhub/assets/122602355/b31cad2b-6294-4d7d-8c4c-7a9ab4aeaf7f)

>![faq](https://github.com/SE-TINF22B6/DHBWhub/assets/122602355/98a2e59c-af5b-4952-b191-8b9b7ab86666)

>![faq-detail](https://github.com/SE-TINF22B6/DHBWhub/assets/122602355/519dac5f-a9f1-4996-8d66-af645fe4eee1)

#### Frontend implementation fixes
We've nailed it this week - our frontend now showcases pictures just the way they should be, no more blank spaces!
>![frontend](https://github.com/SE-TINF22B6/DHBWhub/assets/122602355/5ceb9970-afb4-4bc9-a886-f294db5f41c9)

### Developments on the backend 🧮

* #### Improved security for our API
  We've adapted our CORS configuration and implemented basic access authentication for our endpoints to keep them secure at all times, ensuring that only authorized users can access our data and preventing any unwanted intrusions into our system. These changes fortify our platform's defenses and bolster confidence in our data integrity, giving our users peace of mind as they interact with our services.

* #### Additional endpoints
  Endpoints for the friends page and for the sorting of posts were now finally added, marking a significant milestone in expanding our platform's functionality and enhancing user interaction. With these new endpoints in place, users can seamlessly connect with friends and effortlessly navigate through content!

### Some database fixes 📚 
Quick update on our database: we hustled through some unexpected errors this week by rolling out a new database table. You can now tag your posts without any issues!

### Main goals and priorities in the next weeks 📈  
In the upcoming weeks, we aim to accomplish several goals, which we listed in the last report. Firstly, we strongly aim to establish a connection between the backend and frontend to utilize the database's data. This includes the full integration of the login/registration process and facilitating initial interactions among users on our forum via posts and comments. Additionally, the completion of mockups for the remaining components stays one of our top priorities, providing us clearer visualizations to guide the implementation of new features.

### Sprint 7: What have we achieved ❓ What's to come❔
✅ Frontend: Implement 'Create post'-component
✅ Backend: Adding adjusted models, modelproposals and mapper
✅ Database: Insert example image into database
✅ Database+Backend: Create Tag-Table
✅ Set up Digital Ocean
✅ Backend: Add basic auth for endpoints
✅ Backend: Implement relevant friendship endpoints
✅ Create Mockup for FAQ page
✅ Create Mockup for friends page
🚧 Deploy Frontend to Digital Ocean
🚧 Create Mockup for Saved Posts
🚧 Create Mockup for notifications window
🚧 Create various React components
🚧 Adapt login & signup-dialog
🚧 Improve Backend for authentication (signup/login)
🚧 Remove Herobrine

Until next week 💯

Your DHBWhub Team 🚀

**Comment by NicolasVien** [comment]
Your frontend implementation looks already polished, which is great! I'm looking forward to seeing the new mockups and the friends page come to life. Keep up the good work!

**Comment by mirak55** [comment]
Dear DHBWhub team,
nice work on your frontend, it already looks really awesome. Also nice to see some progress on your backend. I am looking forward to see first working interactions on your  website using the forum.

**Comment by MaarcS** [comment]
Dear DHBWhub team,

I think you did great with your mockups! Your design looks very clean and straightforward, which is optimal for a good user experience. The only thing that I think could be improved, is that the text size varies a lot between the different pages. Fixing that would probably not be a big deal, but it would make your design look more consistent and pleasing for users.

Other than that, great work. It is nice to see that you made a lot of progress.

Your CookHub team

**Comment by Shiny1708** [comment]
I think the new UI looks very clean and polished. Also I'm always a fan of dark mode. I would suggest aligning the Logo with the background of the friend request section. Maybe make it bigger so it covers the whole horizontal space. Security of the API is also very crucial so it's good forward thinking to make it secure. 👍🏼

**Comment by LeonmAndreesen** [comment]
We really like your structure of the post. It is easy to get a picture of what has changed. It is cool to see that you implemented a frequent asked questions in order to gain user satisfaction
